src/gax.ts

```diff
@@ -209,31 +235,50 @@ export class CallSettings {
     let longrunning = this.longrunning;
     let apiName = this.apiName;
     let retryRequestOptions = this.retryRequestOptions;
+
     // If a method-specific timeout is set in the service config, and the retry codes for that
     // method are non-null, then that timeout value will be used to
     // override backoff settings.
-    if (
-      retry !== undefined &&
-      retry !== null &&
-      retry.retryCodes !== null &&
-      retry.retryCodes.length > 0
-    ) {
-      retry.backoffSettings.initialRpcTimeoutMillis = timeout;
-      retry.backoffSettings.maxRpcTimeoutMillis = timeout;
-      retry.backoffSettings.totalTimeoutMillis = timeout;
+    if (retry !== undefined && retry !== null) {
+      // if verify that the retry codes/retry function are not null or undefined
+      if (
+        retry.retryCodesOrShouldRetryFn !== null &&
+        retry.retryCodesOrShouldRetryFn !== undefined
+      ) {
+        // if it's an array of retry codes, make sure it has an element or check if it's a function
+        if (
+          (retry.retryCodesOrShouldRetryFn instanceof Array &&
+            retry.retryCodesOrShouldRetryFn.length > 0) ||
+          retry.retryCodesOrShouldRetryFn instanceof Function
+        ) {
+          retry.backoffSettings.initialRpcTimeoutMillis = timeout;
+          retry.backoffSettings.maxRpcTimeoutMillis = timeout;
+          retry.backoffSettings.totalTimeoutMillis = timeout;
+        }
+      }
     }
+
     // If the user provides a timeout to the method, that timeout value will be used
     // to override the backoff settings.
     if ('timeout' in options) {
       timeout = options.timeout!;
-      if (
-        retry !== undefined &&
-        retry !== null &&
-        retry.retryCodes.length > 0
-      ) {
-        retry.backoffSettings.initialRpcTimeoutMillis = timeout;
-        retry.backoffSettings.maxRpcTimeoutMillis = timeout;
-        retry.backoffSettings.totalTimeoutMillis = timeout;
+      if (retry !== undefined && retry !== null) {
```

Comment:
Is this a duplication? I believe lines 254..256 above do exactly the same.

Reply:
ugh, yes and no, I think it the only difference is what timeout is, but it's def ugly and I can make it better

Reply:
update that confirmed a suspicion I had - it was this duplicate-y [before](https://github.com/googleapis/gax-nodejs/blob/b322f784ec7768e0da781fd54ea5bc9d2347c8cd/src/gax.ts#L212-L238) and I was just maintaining the status quo, but I am pretty sure that this can be fixed with a teeny reorder, so I'll go ahead and do that